core/dbt/include/global_project/macros/materializations/incremental/incremental.sql

```diff
@@ -28,13 +38,32 @@
       {% set build_sql = create_table_as(False, intermediate_relation, sql) %}
       {% set need_swap = true %}
       {% do to_drop.append(backup_relation) %}
+  
   {% else %}
-      {% set tmp_relation = make_temp_relation(target_relation) %}
-      {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+    {% set tmp_relation = make_temp_relation(target_relation) %}
+    {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+    
+    {% if on_schema_change != 'ignore' %}
+      {% set schema_changed = check_for_schema_changes(tmp_relation, target_relation) %}
+      {% do log('schema changed: %s' % schema_changed, info=true) %}
+      {% if schema_changed %}
+        {% do process_schema_changes(on_schema_change, tmp_relation, existing_relation) %}
+        {% set build_sql = incremental_upsert(tmp_relation, target_relation, unique_key=unique_key) %}
+      
+      {% else %}
+        {% do adapter.expand_target_column_types(
```

Comment:
Let's always `expand_target_column_types` before `check_for_schema_changes`, even if it means running a few extra metadata queries, in order to avoid over-aggressively dropping and recreating columns that just need to be varchar/numeric-type expanded

Reply:
Great call - I tested this on Snowflake and works beautifully.